redis/CMakeLists.txt

```diff
@@ -24,7 +24,8 @@ target_sources(redis
     lolwut.cc)
 target_include_directories(redis
   PUBLIC
-    ${CMAKE_SOURCE_DIR}
+    ${CMAKE_SOURCE_DIR})
+target_include_directories(redis SYSTEM
   PRIVATE
     ${scylla_gen_build_dir})
```

Comment:
Marking the entire directory will squash other warnings.

How about renaming protocol_parser.rl to protocol_parser_impl.rl, and add a new protocol_parser.hh that will #include protocol_parser_impl.hh and add pragmas to squelch the warnings?

Reply:
i am adding pragmas in `redis/server.hh` instead. simpler this way.